x/ibc/03-connection/keeper/keeper_test.go

```diff
@@ -171,17 +174,23 @@ func (chain *TestChain) CreateClient(client *TestChain) error {
 	validators := []staking.Validator{validator}
 	histInfo := staking.HistoricalInfo{
 		Header: abci.Header{
+			Time:    client.Header.Time,
 			AppHash: commitID.Hash,
 		},
 		Valset: validators,
 	}
 	client.App.StakingKeeper.SetHistoricalInfo(ctxClient, client.Header.Height, histInfo)
 
+	// also set staking params
+	stakingParams := staking.DefaultParams()
+	stakingParams.HistoricalEntries = 10
+	client.App.StakingKeeper.SetParams(ctxClient, stakingParams)
+
 	// Create target ctx
 	ctxTarget := chain.GetContext()
 
 	// create client
-	clientState, err := ibctmtypes.Initialize(client.ClientID, trustingPeriod, ubdPeriod, client.Header)
+	clientState, err := ibctmtypes.Initialize(chain.ClientID, trustingPeriod, ubdPeriod, client.Header)
```

Comment:
Hmm so looks like my code caused some confusion here.

I believe I should rename this field to `client.ChainID` and document clearly that the clientID that other chains store for the given `client` **is** the chainID of the client chain for simplicity in the tests.

So TestChainA may have a chainID field called: `testchainida`. 

If TestChainB creates a client for TestChainA, it will give it the clientID `testchainida`

Reply:
OK, that's alright too, it was just inconsistent in some places. The key question is why consensus state verification is failing though.

Reply:
Hmm working through the test cases now, and I'm getting a failure on Connection state verification before i hit consensus state verification

Did you comment out the connection state issue to hit consensus state?

Reply:
No, it works for me, can you provide replication instructions?

Reply:
Sorry i meant on my own branch that reverts your identifier issue

Reply:
OK, great! Feel free to revise then.